backends/sqlite/sqlite_backend.go

```diff
@@ -0,0 +1,768 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/acaloiaro/neoq"
+	"github.com/acaloiaro/neoq/handler"
+	"github.com/acaloiaro/neoq/internal"
+	"github.com/acaloiaro/neoq/jobs"
+	"github.com/acaloiaro/neoq/logging"
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	"github.com/iancoleman/strcase"
+	"github.com/jsuar/go-cron-descriptor/pkg/crondescriptor"
+	"github.com/robfig/cron"
+	"golang.org/x/exp/slog"
+)
+
+//go:embed migrations/*.sql
+var sqliteMigrationsFS embed.FS
+
+type contextKey struct{}
+
+const (
+	JobQuery = `SELECT id,fingerprint,queue,status,deadline,payload,retries,max_retries,run_after,ran_at,created_at,error
+					FROM neoq_jobs
+					WHERE id = $1
+					AND status != "processed"`
+	AllPendingJobIDQuery = `SELECT id
+					FROM neoq_jobs
+					WHERE queue = $1
+					AND status != "processed"
+					AND run_after < datetime("now")`
+	FutureJobQuery = `SELECT id,fingerprint,queue,status,payload,retries,max_retries,run_after,ran_at,created_at,error
+					FROM neoq_jobs
+					WHERE queue = $1
+					AND status != "processed"
+					AND run_after > datetime("now")
+					ORDER BY run_after ASC`
+)
+
+var (
+	txCtxVarKey               contextKey
+	ErrNoTransactionInContext = errors.New("context does not have a Tx set")
+	ErrDuplicateJob           = errors.New("duplicate job")
+)
+
+type SqliteBackend struct {
+	neoq.Neoq
+	cancelFuncs       []context.CancelFunc       // cancel functions to be called upon Shutdown()
+	config            *neoq.Config               // backend configuration
+	cron              *cron.Cron                 // scheduler for periodic jobs
+	futureJobs        map[string]*jobs.Job       // map of future job IDs to the corresponding job record
+	handlers          map[string]handler.Handler // a map of queue names to queue handlers
+	queueListenerChan map[string]chan string     // each queue has a listener channel to process enqueued jobs
+	logger            logging.Logger             // backend-wide logger
+	dbMutex           *sync.RWMutex              // protects concurrent access to sqlite db on SqliteBackend
```

Comment:
I was not aware of this mode. I can try experimenting with this and see if it works well.

Reply:
Hi @acaloiaro Let me know what you think and which way to go. I'll do the changes accordingly and wrap this up.

Reply:
1. 

> do we need to wrap entire handleJob code in a transaction (like how it's done in [postgres_backend.go](https://github.com/acaloiaro/neoq/blob/9e3e803ec240a536b7e01abf9c8080dff1b1e52c/backends/postgres/postgres_backend.go#L878))

My recollection of why `handleJob` is wrapped in a transaction is because I was considering exposing access to neoq's transaction to the user, so users always have a transaction available within their jobs. Doing so would ensure that any database modifications within jobs that use the exposed transaction can be rolled back if the job fails. Ultimately, I haven't done any API work to expose the transaction, but I'd like to retain that as an option. So for that reason, I'd like sqlite be able to do the same, if the time comes. I believe there are performance implications to this decision that I haven't fully thought out. If there's a good reason to push back on this decision, I think we should entertain any objections. 

2. Let's not do that. 
3. Ok. Maybe just think through whether we need any settings that are WAL mode. If not, feel free to ignore this. 

Maybe having a mutex isn't a bad idea. We certainly don't want people to have to have special builds of sqlite to have a mutual exclusion guarantee, and I doubt this mutex you've added will dominate performance behavior. I just wanted to make sure it was thoroughly thought through. If having a mutex here achieves the best result in terms of reliability, then we should keep it.

Reply:
Have a look through any remaining, unresolved conversations on the PR. When everything is resolved, I'll give this another review.

Reply:
Also have a look at this top level of this thread. Lints and tests with `-race` are failing. So you'll want to run those locally and get them resolved.